unity/Assets/Scripts/IK_Robot_Arm_Controller.cs

```diff
@@ -517,17 +517,35 @@ public void DropObject()
             jointMeta.rootRelativePosition = FirstJoint.InverseTransformPoint(joint.position);
 
             //local rotation currently relative to immediate parent joint
-            joint.GetChild(0).localRotation.ToAngleAxis(out angleRot, out vectorRot);//getchild to grab the angler since that is what actually changes the geometry angle
-            jointMeta.localRotation = new Vector4(vectorRot.x, vectorRot.y, vectorRot.z, angleRot);
+            if (joint.rotation != joint.GetChild(0).rotation)
+            {
+                joint.GetChild(0).localRotation.ToAngleAxis(out angleRot, out vectorRot);//getchild to grab the angler since that is what actually changes the geometry angle
+                jointMeta.localRotation = new Vector4(vectorRot.x, vectorRot.y, vectorRot.z, angleRot);
+            }
 
+            //edge case for where angler and parent rotations are aligned, which Quaternions have trouble resolving, so we hard-code it here
+            else
+            {
+                jointMeta.localRotation = new Vector4(1, 0, 0, 0);
```

Comment:
I would change the Vector to Quaternion it is more clear, also here I am confused if you want the identity quaternion or one with the `x` component to 1? could you add comment on why this value

Reply:
I used Vector4 because it's not a Quaternion, but the axis of rotation, followed by the angle. That's the format for these orientation outputs. You are correct about x being 1, when technically z should be 1 instead, since it is the forward vector in Unity.

Reply:
localRotation is a Quaternion, if you want an axis of rotation you need to do `Quaternion.AngleAxis(0, new Vector3(1, 0, 0))` I would suggest doing that because what you are doing is setting the raw quaternion values which is represented by 4 real numbers but do not directly mean an Angle Axis rotation, `w` is the real part of the complex number (complex number extension) that a quaternion is and it being zero means a half circle rotation around some axis where `theta` is the angle around that axis and `w = cos(theta / 2)` so `theta` would be `+-π` to satisfy `w = 0` meaning a half circle rotation. localRotation . For `x = 1` in this case it does mean around a unitary vector along x because `sin(theta/2) = 1` with the values noted above, but the general formula is `localRotation.x = axis.x * sin(theta/2)` so it is unintuitive to set the values of the quaternion directly and I am not sure what you're trying to do here

Reply:
The misunderstanding here is in the naming conventions. The name "jointMeta.localRotation" and the transform-property "localRotation" are totally different, albeit with very similar names.

jointMeta.localRotation - A Vector4 where the first three values denote the axis of rotation, and the fourth denotes the number of degrees around that axis. This Quaternion-to-AngleAxis conversion is already done by the line of code above jointMeta.localRotaiton, although I use ToAngleAxis, since AngleAxis has been deprecated.

(transform).localRotation. - A Quaternion denoting a transform's local orientation. Again, this is converted to angle-axis notation before it ever reaches "jointMeta.localRotation".